lib/mayaUsd/ufe/UsdHierarchy.cpp

```diff
@@ -124,12 +124,14 @@ bool UsdHierarchy::hasChildren() const
     //
     // I don't have data that proves we need to worry about performance in here,
     // so going after maintainability.
-    return !children().empty();
+    const bool isFilteringInactive = false;
+    return !createUFEChildList(getUSDFilteredChildren(fItem), isFilteringInactive).empty();
```

Comment:
If we are going with this change then we don't need the comment above anymore. Also have you looked at the changes Krystian made in the past? He was the the one who changed it to use children().

And have you looked at PulledObjectHierarchy and ProxyShapeHierarhcy - they both have similar code.

Reply:
I'll take a look at these classes, thanks.

Reply:
I re-read the comment and it needs to stay: what it cays is that calling create-children-list has extra costs since it will go through all children, but it is simpler to avoid duplicating the logic that filters the items out.